src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -1146,6 +1370,11 @@ def forward(
         # and head_mask is converted to shape [num_hidden_layers x batch x num_heads x seq_length x seq_length]
         head_mask = self.get_head_mask(head_mask, self.config.num_hidden_layers)
 
+        # TODO: maybe have a cleaner way to cast the input (from `Blip2Processor` side?)
+        expected_dtype = self.dtype
+        if encoder_hidden_states is not None and encoder_hidden_states.dtype != expected_dtype:
+            encoder_hidden_states = encoder_hidden_states.to(expected_dtype)
```

Comment:
Is this even necessary?

Reply:
Should not be necessary indeed given that modeling code is by default in torch.float32